### PR TITLE
Switch to Ubuntu docker base image

### PR DESCRIPTION
**Pull request by pugnascotia**
Switch the ES base Docker image for the default and Cloud images to `ubuntu:20.04`, as Ubuntu has a more favourable posture on security updates.

### CODE REVIEW

File: distribution/docker/src/docker/Dockerfile [hunk 1]
```diff
@@ -43,8 +43,13 @@ RUN chmod 0555 /bin/tini
 ################################################################################
 FROM ${base_image} AS builder
 
+<% if (docker_base == 'default' || docker_base == 'cloud') { %>
+# Install required packages to extract the Elasticsearch distribution
```
Comment (mark-vieira) [inline distribution/docker/src/docker/Dockerfile:47]: We can move this comment outside this conditional so we don't have to duplicate it, yes?

**Review by mark-vieira — APPROVED** [review]
LGTM

**Comment by pugnascotia** [comment]
@elastic/cloud-delivery the ES Delivery team is considering moving our base image to Ubuntu 21.04, in order to benefit from what we understand is a better security posture w.r.t. CVEs. How can we determine the impact to Cloud? The most obvious source of breaks are CLI tool incompatibilities.

cc @jasontedor

**Comment by pugnascotia** [comment]
@elasticsearchmachine run elasticsearch-ci/docs

**Comment by pugnascotia** [comment]
@elasticsearchmachine run elasticsearch-ci/packaging-tests-unix 
@elasticsearchmachine run elasticsearch-ci/part-2

**Comment by alpar-t** [comment]
@pugnascotia This will need some work on the cloud side in 7.x for sure, see https://github.com/elastic/cloud-assets/blob/7.x/docker/elasticsearch/cloudimage/Dockerfile#L13 . In 8.0.0 we are no longer building any docker image on the cloud side, so at least from the perspective of docker images we are good there. 

The second thing to look at is https://github.com/elastic/cloud-assets/blob/7.x/stackpack/elasticsearch/elasticsearch.sh and https://github.com/elastic/cloud-assets/blob/7.x/stackpack/elasticsearch/on-error-exitcode.sh we still use these for 8.0.0 too. 
We don't use anything exotic as part of these, but we do use some tools that we need to make sure are there like unzip and wget. I would be surprised if we used some options on the tools that would cause problems between distributions, especially that my understanding is that we're potentially using newer versions and these have a good track record of backward comparability.  

The ultimate way to be sure would be to somehow deploy the image to `elasticsearch-ci` and create a deployment in the CFT region, then exercise some cloud features on it, like installing a custom plugin, user bundle etc to make sure anything that's done in these scripts is covered.  I think it would be worth having that as an automated way of validating changes to the ES docker image.

**Comment by mark-vieira** [comment]
> The ultimate way to be sure would be to somehow deploy the image to elasticsearch-ci and create a deployment in the CFT region, then exercise some cloud features on it, like installing a custom plugin, user bundle etc to make sure anything that's done in these scripts is covered. I think it would be worth having that as an automated way of validating changes to the ES docker image.

This isn't going to be in place immediately. What's our way forward for trying to get this merged for 7.16? We want a solution here as our current Docker image build procedure is blocking us from being published on Dockerhub, and we'd like to sort this out for 7.16.

**Comment by pugnascotia** [comment]
I talked to the Cloud Delivery team, and actually the situation now is that we need to coordination with the Cloud Applications team, so I talked to them too.

The infrastructure in the Cloud-First Testing (CFT) region is ready to go, provided we can push the image we wan to run under `docker.elastic.co/elasticsearch-ci/elasticsearch-cloud:$LABEL`. There are issues, however:

   1. There are snapshot builds for that pattern in the Elastic registry, but none recent
   2. It's not obvious that any of them correspond to PRs - I don't believe they do

I'm trying to find out why they're out-of-date, then we need to figure out how to push images during PR builds. By definition, that needs to be separate from the `release-manager` builds, since the code changes aren't merged.

We could make progress with this PR by merging and then testing a snapshot, but there's no point doing that until (1) is resolved.

**Comment by pugnascotia** [comment]
I solved (1) - the images are published to [cloud-release/elasticsearch-cloud](https://container-library.elastic.co/r/cloud-release/elasticsearch-cloud?show_snapshots=true). See https://github.com/elastic/infra/issues/32386#issuecomment-948003634 for background.

**Comment by pugnascotia** [comment]
@mark-vieira for (2) above, what do you think about adding support to the build for pushing a Docker image? I'm not sure in which phase of the CI build we'd put that - part of `packaging-tests-unix` / `packaging-tests-unix-sample` maybe?

We'd also need to provide / configure registry credentials. Maybe `.ci/init.gradle` could do that?

**Comment by mark-vieira** [comment]
> @mark-vieira for (2) above, what do you think about adding support to the build for pushing a Docker image? I'm not sure in which phase of the CI build we'd put that - part of packaging-tests-unix / packaging-tests-unix-sample maybe?

Is this is prerequisite for getting this change through? We can certainly get a PR job added that pushes an image, one thing I haven't worked out yet is what we're going to use for naming conventions for the tags. I presume the PR number would have to be part of it so that folks don't clobber eachothers images.

**Comment by pugnascotia** [comment]
Well, only CI is supposed to have push access to the `<product>-ci` Docker repo namespaces, and we need to do this work anyway for CFT, so... 🤷 

Apparently Cloud pushes images with the commit hash in the tag, but I think I'd prefer to use the PR number as you suggest. We already include the commit as a label on the image if someone needs that. So it would be e.g:

    docker.elastic.co/elasticsearch-ci/elasticsearch-cloud-ess:8.0.0-prXXXXX

(I figure there's no point having a `-SNAPSHOT` suffix in there, given the PR suffix)

**Comment by pugnascotia** [comment]
@mark-vieira I had a go at implementing something, see what you think. I took a guess at the Jenkins stuff 😅

**Comment by pugnascotia** [comment]
Mark and I deployed the resulting image in the Cloud-First Testing region successfully, so we're good to merge for 8.1 and 8.0. The backport to 7.16 will require coordination with Cloud.

**Comment by pugnascotia** [comment]
Switch from CentOS to Ubuntu backported via a series of cherry-picks:

   1. d8a827409da
   2. db89cabed11
   3. 2febd2f401e
   4. e9bfea5d1d1

**Pull request merged**